src/components/autocomplete/js/autocompleteController.js

```diff
@@ -529,6 +529,7 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
       if (searchText !== $scope.searchText) return; //-- just cache the results if old request
       ctrl.matches = matches;
       ctrl.hidden  = shouldHide();
+      if ($scope.selectOnMatch) selectItemOnMatch();
```

Comment:
The user only provides a flag, not the function itself.  No need for `angular.noop`, unless I am misunderstanding your suggestion.